mayastor/src/bdev/nexus/nexus_child.rs

```diff
@@ -435,7 +435,7 @@ impl NexusChild {
                 key,
                 0,
                 nvme_reservation_acquire_action::ACQUIRE,
-                nvme_reservation_type::WRITE_EXCLUSIVE,
+                nvme_reservation_type::WRITE_EXCLUSIVE_ALL_REGS,
```

Comment:
Do we always want WRITE_EXCLUSIVE_ALL_REGS? Would there be a case for WRITE_EXCLUSIVE, eg to ensure only 1 nexus can access the replicas?

Reply:
The existing code does only allow 1 nexus to access the replicas, and adding the replicas to a new nexus pre-empts the old nexus and prevents it from writing the replicas. My understanding is we want to move away from that behaviour and allow more than 1 nexus to connect to a replica. If we need to maintain 2 different behaviours, then a flag needs to be added to the Nexus structure to signal that and a new field in NexusCreateV2, or perhaps using V2 to signal a "HA" nexus and V1 for the old single-writer Nexus.

Reply:
Ok, it seems to be a bit "risky" to allow all nexus to access the replicas concurrently unless we have a use case and solution for doing RR, but I'm not in the loop of what we're aiming for here so maybe I need to revisit the ANA docs.

Reply:
At the moment any nexus, or indeed any NVMe client, can access any replica anyway, unless the `NEXUS_NVMF_RESV_ENABLE` environment variable is set.